src/stores/createUploadObservationsSlice.ts

```diff
@@ -0,0 +1,153 @@
+const DEFAULT_STATE = {
+  uploadError: null,
+  // Single error caught during multiple obs upload
+  multiError: null,
+  errorsByUuid: {},
+  uploaded: [],
+  singleUpload: true,
+  uploads: [],
+  numToUpload: 0,
+  // Increments even if there was an error, so here "finished" means we tried
+  // to upload it, not that it succeeded
+  numFinishedUploads: 0,
+  totalToolbarProgress: 0,
+  totalUploadProgress: [],
+  uploadStatus: "pending",
+  totalToolbarIncrements: 0
+};
+
+interface TotalUploadProgress {
+  uuid: string,
+  totalIncrements: number,
+  currentIncrements: number,
+  totalProgress: number
+}
+
+interface UploadObservationsSlice {
+  uploadError: string | null,
+  multiError: string | null,
+  errorsByUuid: Object,
+  uploaded: Array<string>,
+  singleUpload: boolean,
+  uploads: Array<Object>,
+  numToUpload: number,
+  numFinishedUploads: number,
+  totalToolbarProgress: number,
+  totalUploadProgress: Array<TotalUploadProgress>,
+  uploadStatus: string | null,
```

Comment:
Here too. `uploadStatus` has a finite set of possible values, so they should be strongly typed.  If someone tries to set `uploadStatus` to `"whatever"` we should get an error